joern-cli/frontends/swiftsrc2cpg/src/main/scala/io/joern/swiftsrc2cpg/passes/ExtensionCallPass.scala

```diff
@@ -0,0 +1,49 @@
+package io.joern.swiftsrc2cpg.passes
+
+import io.joern.x2cpg
+import io.shiftleft.codepropertygraph.generated.{Cpg, DispatchTypes, PropertyNames}
+import io.shiftleft.codepropertygraph.generated.nodes.*
+import io.shiftleft.passes.ForkJoinParallelCpgPass
+import io.shiftleft.semanticcpg.language.*
+
+/** Pass that resolves extension-method-style calls and rewrites them to use static dispatch to the corresponding
+  * extension method fullName.
+  *
+  * The pass scans call nodes in the provided CPG and for each call methodFullName present in `extensionFullNameMapping`
+  * it:
+  *   - updates the call's dispatch type to STATIC_DISPATCH
+  *   - replaces the call's methodFullName with the mapped extension method full name
+  *
+  * @param cpg
+  *   the code property graph to operate on
+  * @param extensionFullNameMapping
+  *   mapping from compiler generated methodFullName -> CPG extension methodFullName (for uniqueness)
+  */
+class ExtensionCallPass(cpg: Cpg, extensionFullNameMapping: Map[String, String])
```

Comment:
I think the approach of fixing up the call sites later has reached its end because as mentioned about the receiver edges are wrong for static calls. Of course we could now remove them here but a pre-pass which collects the necessary information to directly generate the correct calls seems more appropriate.